### PR TITLE
PHPCS: root functions

### LINKED ISSUE #21953: PHPCS: Root function files
### Impacted plugin

Jetpack

### What

PHPCS

### How

```
"projects/plugins/jetpack/functions.compat.php",
	"projects/plugins/jetpack/functions.gallery.php",
	"projects/plugins/jetpack/functions.photon.php",
"projects/plugins/jetpack/json-endpoints.php",
"projects/plugins/jetpack/locales.php",
"projects/plugins/jetpack/require-lib.php",
```

### PR DESCRIPTION
**Pull request by bindlegirl**
- PHPCS changes
- Add changelog

Fixes #21953

#### Changes proposed in this Pull Request:

* PHPCS fixing

#### Jetpack product discussion

* Changes will be required for wp.com files. Discussion: p1650553854321699-slack-C02HQGKMFJ8
* Adding new a new file class-jetpack-gallery-settings.php to replace the functions.gallery.php is also required. Once this PR is reviewed, this will be handled in a separate PR.

#### Does this pull request change what data or activity we track or use?

no
#### Testing instructions:

* Check if galleries work.
* Are tests failing?

### CODE REVIEW

File: projects/plugins/jetpack/functions.compat.php [hunk 1]
```diff
@@ -27,71 +33,95 @@ function jetpack_get_youtube_id( $url ) {
...
  *
  * @see WPCOM/wp-content/mu-plugins/string-helpers.php
  */
 if ( ! function_exists( 'wp_startswith' ) ) :
+	/**
+	 * Filters strings that start with a specific substring.
+	 *
+	 * @param var    $haystack String we are filtering.
+	 * @param string $needle The substring we are looking for.
+	 */
```
Comment (anomiex) [inline projects/plugins/jetpack/functions.compat.php:91]: May as well document the return values when we're adding doc comments.
```suggestion
	 * @return bool
	 */
```

File: projects/plugins/jetpack/functions.compat.php [hunk 2]
```diff
@@ -27,71 +33,95 @@ function jetpack_get_youtube_id( $url ) {
...
 
 /**
- * Merge in three string helper functions from WPCOM.
+ * Merge in three string helper functions from WPCOM to make working with strings easier.
  *
  * @see WPCOM/wp-content/mu-plugins/string-helpers.php
  */
 if ( ! function_exists( 'wp_startswith' ) ) :
+	/**
+	 * Filters strings that start with a specific substring.
```
Comment (anomiex) [inline projects/plugins/jetpack/functions.compat.php:87]:
```suggestion
	 * Check whether a string starts with a specific substring.
```

File: projects/plugins/jetpack/functions.compat.php [hunk 3]
```diff
@@ -27,71 +33,95 @@ function jetpack_get_youtube_id( $url ) {
...
+	 */
 	function wp_startswith( $haystack, $needle ) {
 		return 0 === strpos( $haystack, $needle );
 	}
 endif;
 
-
 if ( ! function_exists( 'wp_endswith' ) ) :
+	/**
+	 * Filters strings that end with a specific substring.
```
Comment (anomiex) [inline projects/plugins/jetpack/functions.compat.php:99]:
```suggestion
	 * Check whether a string ends with a specific substring.
```

File: projects/plugins/jetpack/functions.compat.php [hunk 4]
```diff
@@ -27,71 +33,95 @@ function jetpack_get_youtube_id( $url ) {
...
+	 *
+	 * @param var    $haystack String we are filtering.
+	 * @param string $needle The substring we are looking for.
+	 */
```
Comment (anomiex) [inline projects/plugins/jetpack/functions.compat.php:103]:
```suggestion
	 * @return bool
	 */
```

File: projects/plugins/jetpack/functions.compat.php [hunk 5]
```diff
@@ -27,71 +33,95 @@ function jetpack_get_youtube_id( $url ) {
...
 	function wp_endswith( $haystack, $needle ) {
-		return $needle === substr( $haystack, -strlen( $needle ));
+		return $needle === substr( $haystack, -strlen( $needle ) );
 	}
 endif;
 
 if ( ! function_exists( 'wp_in' ) ) :
+	/**
+	 * Filters strings that contain a specific substring.
```
Comment (anomiex) [inline projects/plugins/jetpack/functions.compat.php:111]:
```suggestion
	 * Checks whether a string contains a specific substring.
```

File: projects/plugins/jetpack/functions.compat.php [hunk 6]
```diff
@@ -27,71 +33,95 @@ function jetpack_get_youtube_id( $url ) {
...
+	 *
+	 * @param string $needle The substring we are looking for.
+	 * @param var    $haystack String we are filtering.
+	 */
```
Comment (anomiex) [inline projects/plugins/jetpack/functions.compat.php:115]:
```suggestion
	 * @return bool
	 */
```

**Review by anomiex — COMMENTED** [review]
A few suggestions on doc comments, otherwise looks sensible to me.

**Review by kraftbj — APPROVED** [review]

**Comment by kraftbj** [comment]
Fixing the WP.com side now.

**Comment by kraftbj** [comment]
r244453-wpcom

**Pull request merged**